### PR TITLE
Moving away from msdia managed wrapper, to doing P/Invoke on msdia.

### PR DESCRIPTION
**Pull request by mayankbansal018**
###Test Matrix

|           | TestHost SDK(Old)                                          | TestHost SDK(New)                                 |
|-----------|------------------------------------------------------------| --------------------------------------------------|
VsTest(Old) | Portable-- No   Change   Full - No Change                  | Portable --   Navigation Fixed   Full -- No Change|
VsTest(New) | Portable--   Navigation Fail(No Change)   Full - No Change | Portable--   Navigation Fixed   Full - No Change|

### CODE REVIEW

**Comment by mayankbansal018** [comment]
@dotnet-bot test this please

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs [hunk 1]
```diff
@@ -128,6 +112,57 @@ public INavigationData GetNavigationData(string declaringTypeName, string method
...
+                    if (string.IsNullOrEmpty(searchPath))
+                    {
+                        searchPath = Directory.GetCurrentDirectory();
+                    }
+                }
+
+                // Load the data for the executable
+                if (HResult.Failed(this.source.LoadDataForExe(filename, searchPath, IntPtr.Zero)))
+                {
+                    return false;
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs]: throw exception with message here.

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs [hunk 2]
```diff
@@ -138,14 +173,29 @@ private DiaNavigationData GetSymbolNavigationData(IDiaSymbol symbol)
 
             try
             {
-                this.session.findLinesByAddr(symbol.addressSection, symbol.addressOffset, (uint)symbol.length, out lines);
+                // Get the address section
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs]: Should we combined all three 'if' statements with `||`.

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs [hunk 3]
```diff
@@ -0,0 +1,651 @@
...
+    }
+
+    internal static class DiaSourceObject
+    {
+        [DllImport("kernel32.dll", SetLastError = true)]
+        public static extern IntPtr LoadLibraryEx(string lpFileName, IntPtr hReservedNull, int dwFlags);
+
+        public static IDiaDataSource GetDiaSourceObject()
+        {
+            var currentDirectory = new ProcessHelper().GetCurrentProcessLocation();
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs]: Check how we are handling `/Inproc` scenario.

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs [hunk 4]
```diff
@@ -0,0 +1,651 @@
...
+
+            IntPtr modHandle = IntPtr.Zero;
+            if (IntPtr.Size == 8)
+            {
+                modHandle = LoadLibraryEx(Path.Combine(currentDirectory, "x64\\msdia140.dll"), IntPtr.Zero, 0);
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs]: Should we Path.PathSeparator here?

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs [hunk 5]
```diff
@@ -0,0 +1,651 @@
...
+            }
+            else
+            {
+                modHandle = LoadLibraryEx(Path.Combine(currentDirectory, "x86\\msdia140.dll"), IntPtr.Zero, 0);
+            }
+
+            if(modHandle == IntPtr.Zero)
+            {
+                // Failed to load msdia140, why?
+                return null;
```
Comment (smadala) [inline src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs]: Log message or throw here.

File: src/Microsoft.TestPlatform.PlatformAbstractions/netcore/System/ProcessHelper.cs [hunk 6]
```diff
@@ -0,0 +1,18 @@
...
+namespace Microsoft.VisualStudio.TestPlatform.PlatformAbstractions
+{
+    using System.IO;
+    using System.Reflection;
+    using Microsoft.VisualStudio.TestPlatform.PlatformAbstractions.Interfaces;
+
+    public partial class ProcessHelper : IProcessHelper
+    {
+        /// <inheritdoc/>
+        public string GetCurrentProcessLocation()
```
Comment (smadala) [inline src/Microsoft.TestPlatform.PlatformAbstractions/netcore/System/ProcessHelper.cs]: Method name won't match defination, Should we rename to EntryPointDirectory?

**Review by smadala — APPROVED** [review]

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs (hunk 3 above)
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs]: Will check this once InProc comes into play. We will have to make some changes then definitely, will let faizan know

File: src/Microsoft.TestPlatform.PlatformAbstractions/netcore/System/ProcessHelper.cs (hunk 6 above)
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.PlatformAbstractions/netcore/System/ProcessHelper.cs]: from users perspective in netcore, the process would be the dll he/she is creating.

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs (hunk 2 above)
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs]: we can, but I think code will be less readable.

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs (hunk 5 above)
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Navigation/NativeMethods.cs]: done

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs (hunk 1 above)
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs]: done

File: src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs [hunk 7]
```diff
@@ -35,16 +31,11 @@ internal class FullSymbolReader : ISymbolReader
...
         /// Methods in different types can have same name, hence seprated dicitionary is created for each type.
         /// Bug: Method overrides in same type are not handled (not a regression)
+        /// ToDo(Solution): Use method token along with as identifier, this will always give unique method.The adapters would need to send this token to us to correct the behavior.
         /// </summary>
         private Dictionary<string, Dictionary<string, IDiaSymbol>> methodSymbols = new Dictionary<string, Dictionary<string, IDiaSymbol>>();
 
         /// <summary>
-        /// Manifest files for Reg Free Com. This is essentially put in place for the msdia dependency.
-        /// </summary>
-        private const string ManifestFileNameX86 = "TestPlatform.ObjectModel.x86.manifest";
```
[the remark this answers is not in the document]
Comment (mayankbansal018) [reply src/Microsoft.TestPlatform.ObjectModel/Navigation/FullSymbolReader.cs]: done

**Pull request merged**